src/NuGet.Core/NuGet.PackageManagement/Projects/ProjectJsonBuildIntegratedNuGetProject.cs

```diff
@@ -329,10 +348,16 @@ private JObject GetJson()
 
         private void UpdateFramework(JObject json)
         {
+            // Update the internal target framework with TPMinV from csproj
+            UpdateInternalTargetFramework();
+
             var frameworks = JsonConfigUtility.GetFrameworks(json);
-            if (InternalMetadata.TryGetValue(NuGetProjectMetadataKeys.TargetFramework, out object newTargetFramework))
+            if (InternalMetadata.TryGetValue(NuGetProjectMetadataKeys.TargetFramework, out object newTargetFrameworkObject))
```

Comment:
To avoid duplicated code, consider adding a helper method like `TryGetInternalTargetFramework`.